web-app/js/portal/cart/DownloadConfirmationWindow.js

```diff
@@ -115,6 +110,15 @@ Portal.cart.DownloadConfirmationWindow = Ext.extend(Ext.Window, {
         this.onAcceptCallback = params.onAccept;
 
         Portal.cart.DownloadConfirmationWindow.superclass.show.call(this);
+
+        var metadataRecord = params.collection.getMetadataRecord();
```

Comment:
@markhepburn Some mocking required for that in unit tests.

Not going to debate the quality of the unit tests just quite yet, but... You can probably plug it in an adequate way.